src/ipc/dbus/DBusObjectRegistry.cpp

```diff
@@ -78,30 +89,52 @@ void DBusObjectRegistry::unregisterObject(const QString &objectPath)
 {
     init();
     auto serviceName = DBusManager::instance().serviceName();
-    if (m_master) {
+    if (isMaster()) {
         m_master->unregisterObject(objectPath, serviceName);
     } else {
         return m_objectRegistryAsyncProxy->unregisterObject(objectPath, serviceName);
     }
 }
 
-const QMap<QString, QString> &DBusObjectRegistry::objects(bool blocking)
+QMap<QString, QString> DBusObjectRegistry::objects(bool blocking)
 {
     init();
-    if (m_master) {
-        return m_master->objects();
-    } else {
-        if (blocking) {
+    if (!isMaster() && blocking) {
+        if (m_objects.isEmpty()) {
```

Comment:
you mean "if (registry already received by the async proxy)", right ? That registry can be empty, even if it has been received.
Please fix.